### PR TITLE
Updates to ClamAV

### PR DESCRIPTION
**Pull request by jcbrand**
- Update ClamAV servcices from upstart "jobs" to systemd.service. 
- Add new service for `clamd`, the ClamAV daemon.

CC and thanks to @cillianderoiste

### CODE REVIEW

**Comment by jagajaga** [comment]
Can you please squash all your commits and name the result as `clamav: improvements` and also add notes about your improvements in that commit's message.

File: nixos/modules/services/security/clamav.nix [hunk 1]
```diff
@@ -3,78 +3,124 @@ with lib;
...
+  '';
 in
 {
-  ###### interface
-
   options = {
-
     services.clamav = {
+      daemon = {
+        enable = mkOption {
```
Comment (jagajaga) [inline nixos/modules/services/security/clamav.nix]: Use `mkEnableOption` please.

File: nixos/modules/services/security/clamav.nix [hunk 2]
```diff
@@ -3,78 +3,124 @@ with lib;
...
+        };
+      };
       updater = {
-	enable = mkOption {
-	  default = false;
-	  description = ''
-	    Whether to enable automatic ClamAV virus definitions database updates.
-	  '';
-	};
+        enable = mkOption {
```
Comment (jagajaga) [inline nixos/modules/services/security/clamav.nix]: Use `mkEnableOption` please.

**Comment by jcbrand** [comment]
hi @jagajaga 

The commits are squashed but I kept the `Fix indentation` commit separate, to keep the diffs between different revisions more readable. It gets difficult to see exactly which changes were made if indentation changes are also included in a diff.

The code is now also using `mkEnableOption`.

File: nixos/modules/services/security/clamav.nix [hunk 3]
```diff
@@ -3,78 +3,121 @@ with lib;
...
+  '';
 in
 {
-  ###### interface
-
   options = {
-
     services.clamav = {
+      daemon = {
+        enable = mkEnableOption {
```
Comment (jagajaga) [inline nixos/modules/services/security/clamav.nix]: It's not correct. You just need to write `enable = mkEnableOption "clamd daemon";`.

File: nixos/modules/services/security/clamav.nix [hunk 4]
```diff
@@ -3,78 +3,121 @@ with lib;
...
+        };
+      };
       updater = {
-	enable = mkOption {
-	  default = false;
-	  description = ''
-	    Whether to enable automatic ClamAV virus definitions database updates.
-	  '';
-	};
+        enable = mkEnableOption {
```
Comment (jagajaga) [inline nixos/modules/services/security/clamav.nix]: Same here.

**Comment by jcbrand** [comment]
@jagajaga ok, like now?

**Comment by jagajaga** [comment]
LGTM. Have you tested the latest version of your improvements?

**Comment by jcbrand** [comment]
@jagajaga I just tested it again and looks good.

**Pull request merged**